epochStart/metachain/rewards.go

```diff
@@ -546,3 +556,25 @@ func (rc *rewardsCreator) RemoveBlockDataFromPools(metaBlock *block.MetaBlock, b
 			"num txs", mbHeader.TxCount)
 	}
 }
+
+//TODO evaluate if this function will remain here or be moved inside a new rewardsCreator implementation
+// in case it should remain here, we need a softfork protection for this call
+func (rc *rewardsCreator) prepareRewardsFromStakingSC(validatorsInfo map[uint32][]*state.ValidatorInfo) error {
```

Comment:
this function name is prepareRewards -> under it you call ComputeRewards - and after this function is called we have the real rewards computation. these names are confusing

Reply:
will change

Reply:
renamed to `prepareRewards`

Reply:
this is not necessarily preparing rewards - it prepares staking data.

Reply:
renamed to `prepareStakingData`